File: src/LightNap.Core/StaticContents/Dto/Response/PublishedStaticContentResultDto.cs
using LightNap.Core.StaticContents.Models;

namespace LightNap.Core.StaticContents.Dto.Response
{
    /// <summary>
    /// The result of a request for a specific CMS item based on user permissions.
    /// </summary>
    public class PublishedStaticContentResultDto
    {
        /// <summary>
        /// The visibility state of the requested item for the current user.
        /// </summary>
        public required StaticContentUserVisibility Visibility { get; set; }

        /// <summary>
        /// The CMS item.
        /// </summary>
        public PublishedStaticContentDto? Content { get; set; }
    }
}

File: src/lightnap-ng/src/app/core/backend-api/dtos/static-contents/response/published-static-content-result-dto.ts
import { StaticContentUserVisibility } from "@core/backend-api/static-content-user-visibility";
import { PublishedStaticContentDto } from "./published-static-content-dto";

/**
 * The result of published static content request.
 */
export interface PublishedStaticContentResultDto {
    /**
     * The visibility for the current user.
     */
    visibility: StaticContentUserVisibility;

    /**
     * The content.
     */
    content?: PublishedStaticContentDto;
}
